libr/include/r_util/r_str.h

```diff
@@ -23,12 +24,38 @@ typedef int (*RStrRangeCallback) (void *, int);
 
 static inline void r_str_rmch(char *s, char ch) {
 	for (;*s; s++) {
-		if (*s==ch) {
+		if (*s == ch) {
 			memmove (s, s + 1, strlen (s));
 		}
 	}
 }
 
+//  -- strf
+typedef struct r_strf_t {
+	size_t idx, size;
+	char buf[0];
+} r_strf__;
+
+R_NULLABLE static inline char *r_strf_(r_strf__ *s, const char *fmt, ...) {
+	if (s->idx >= s->size) {
+		s->idx = 0;
+	}
+	r_return_val_if_fail (s && fmt, NULL);
+	va_list ap;
+	va_start (ap, fmt);
+	char *p = s->buf + s->idx;
+	size_t left = s->size - s->idx - 1;
+	int res = vsnprintf (p, left, fmt, ap);
+	r_return_val_if_fail (res != left, NULL);
+	s->idx += res + 1;
+	va_end (ap);
+	return p;
+}
+
+#define r_strf_frame(x) struct r_strf_t { size_t idx, size; char buf[x]; } r_strf_var = {0, x, 0}
```

Comment:
no need to give the struct name, I think. We can keep it anonymous. Moreover, this `r_strf_t` is different from what defined above.

Reply:
@radare just use anonymous structs. The struct you are defining here is:
```C
struct r_strf_t {
  size_t idx, size;
  char buf[x];
}
```
but then you are actually using another struct which is:
```C
struct r_strf_t {
  size_t idx, size;
  char buf[0];
}
```

Since you don't need the first structure's name, just make it anonymous.

(please try to answer in the comments below mine, otherwise the review becomes a mess).

Reply:
thats what happens when i reply from mail, it was working before it was bought my microsoft... so anyways, im using the web now.

Reply:
i know, and the reason for having [0] is because its an empty field, which means the struct size wont handle this field of any size and it is actually correct

Reply:
I know! I'm just saying don't use the struct name r_strf_t in this line because it is useless. And it is anyway different from the same `struct r_strf_t` defined above (as you said, one uses [0], the other does not).